### PR TITLE
feat(storybook): Co-habitating/Upgrading Storybooks to v7 (dependency madness ensues)

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

This PR does the following:
* ~~Temporarily removes the deny-list for license types. For some reason it's blocking a goofily-licensed submodule, which is perfectly license compatible (MIT). I'll flip this back on once I'm done with this effort.~~ Resolved.
* Upgrades BOTH the Superset Storybook and superset-ui-demo to Storybook 7
* Migrates all the Stories to use Controls instead of Knobs
* Includes all the Stories from superset-ui-demo in the main Superset storybook (you can still run `superset-ui-demo` on its own as well, though)
* Migrates `mdx` stories (overviews) to `tsx` - these might need touchups
* Fixes a bunch of little random bugs along the way
* Removes a bunch of package vulnerabilities (1 critical, 17 high, 11 moderate)
* Bumps ESLint and Prettier by necessity, and for the sake of making this PR mergable, disables the new offences - we'll get those in follow-up PRs, I promise.
* Touches many other moving parts - it's all a house of cards... but things seem to be running stably. 

### TESTING INSTRUCTIONS

Run the root storybook and look at all the bells and whistles!

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -22,4 +22,3 @@ jobs:
           fail-on-severity: critical
           # compatible/incompatible licenses addressed here: https://www.apache.org/legal/resolved.html
           # find SPDX identifiers here: https://spdx.org/licenses/
-          deny-licenses: MS-LPL, BUSL-1.1, QPL-1.0, Sleepycat, SSPL-1.0, CPOL-1.02, AGPL-3.0, GPL-1.0+, BSD-4-Clause-UC, NPL-1.0, NPL-1.1, JSON
```
Comment (mistercrunch) [inline .github/workflows/dependency-review.yml:25]: did you mean to delete this?

Comment (rusackas) [reply .github/workflows/dependency-review.yml:25]: Yes (mentioned in the PR description), I was hoping to revisit in a follow-up PR. I'll throw some more attempts at it to see if I can fix it here.

**Review by mistercrunch — APPROVED** [review]

**Review by eschutho — APPROVED** [review]
Wow! 🎉

**Comment by mistercrunch** [comment]
MERGE QUICK #yolo 
giphy (1)

**Comment by rusackas** [comment]
Just need a final code-owner check from @michael-s-molina @kgabryje or @geido (because I touched Select, MetadataBar, DropdownContainer) and we'll be good to smash this thing in!

File: superset-frontend/.eslintrc.js [hunk 2]
```diff
@@ -161,6 +162,28 @@ module.exports = {
         'react/static-property-placement': 0, // re-enable up for discussion
         'prettier/prettier': 'error',
         'file-progress/activate': 1,
+        // delete me later: temporary rules to help with migration
```
Comment (geido) [inline superset-frontend/.eslintrc.js:165]: Tracking this here https://github.com/orgs/apache/projects/315/views/1?pane=issue&itemId=53360236

File: superset-frontend/src/components/Table/cell-renderers/ActionCell/ActionCell.overview.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// import { Meta, Source, Story, ArgsTable } from '@storybook/addon-docs';
```
Comment (geido) [inline superset-frontend/src/components/Table/cell-renderers/ActionCell/ActionCell.overview.tsx:20]:
```suggestion
```

File: superset-frontend/src/components/Table/cell-renderers/ActionCell/ActionCell.overview.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import Markdown from 'markdown-to-jsx';
+// import { ActionMenuItem } from 'src/components/Table/cell-renderers/index';
+// import ActionCell from './index';
```
Comment (geido) [inline superset-frontend/src/components/Table/cell-renderers/ActionCell/ActionCell.overview.tsx:23]:
```suggestion
```

File: superset-frontend/webpack.config.js [hunk 5]
```diff
@@ -445,24 +444,20 @@ const config = {
...
+      //       loader: require.resolve('@storybook/mdx2-csf/loader'),
+      //       options: {
+      //         skipCsf: false,
+      //         mdxCompileOptions: {
+      //           remarkPlugins: [remarkGfm],
+      //         },
+      //       },
+      //     },
+      //   ],
+      // },
```
Comment (geido) [inline superset-frontend/webpack.config.js:460]:
```suggestion
```

File: superset-frontend/.eslintrc.js [hunk 6]
```diff
@@ -310,6 +351,24 @@ module.exports = {
     'react/static-property-placement': 0, // disabled temporarily
     'react-prefer-function-component/react-prefer-function-component': 1,
     'prettier/prettier': 'error',
+    // disabling some things that come with the eslint 7->8 upgrade. Will address these in a separate PR
```
Comment (geido) [inline superset-frontend/.eslintrc.js:354]: Tracked here as well https://github.com/orgs/apache/projects/315/views/1?pane=issue&itemId=53360236

**Review by geido — APPROVED** [review]
Thanks for all the great work that went into this! Let's keep an eye on Netlify to make sure everything goes smooth when we merge this up.

**Review by kgabryje — APPROVED** [review]
Nothing to add on top of Diego’s comments, amazing work!!

**Review by michael-s-molina — COMMENTED** [review]
Thank you for all the hard work @rusackas!

While testing the PR, I found the following problems:

Some components are not rendering anymore:

Screenshot 2024-02-15 at 08 52 26

Screenshot 2024-02-15 at 08 57 48
<br>
Some components are rendering but their content is not:

Screenshot 2024-02-15 at 08 55 05
<br>
Some controls that used to work are not rendering. In this example, sizes and styles were a select:

Screenshot 2024-02-15 at 08 56 31
<br>
Some plugins are overflowing the render area:

Screenshot 2024-02-15 at 08 58 45
Screenshot 2024-02-15 at 08 59 06

This list is not exhaustive, as I only tested a sample of the stories. I can confirm that these problems don't happen on master. Given the size of the PR and the possibility of conflicts, I'm ok with merging it as is and tackling the problems in a follow-up given that the Storybook is an isolated area.

Amazing work 👏🏼

**Comment by rusackas** [comment]
Follow-up PRs to address all of the above coming soon! I was spot checking stories as I ported them to the new format, but knew that some would get screwed up along the way. Let the polishing phase begin!

**Comment by geido** [comment]
> Thank you for all the hard work @rusackas!
> 
> While testing the PR, I found the following problems:
> 
> Some components are not rendering anymore:
> 
> Screenshot 2024-02-15 at 08 52 26 Screenshot 2024-02-15 at 08 57 48
> Some components are rendering but their content is not: Screenshot 2024-02-15 at 08 55 05
> Some controls that used to work are not rendering. In this example, sizes and styles were a select: Screenshot 2024-02-15 at 08 56 31
> Some plugins are overflowing the render area: Screenshot 2024-02-15 at 08 58 45 Screenshot 2024-02-15 at 08 59 06
> This list is not exhaustive, as I only tested a sample of the stories. I can confirm that these problems don't happen on master. Given the size of the PR and the possibility of conflicts, I'm ok with merging it as is and tackling the problems in a follow-up given that the Storybook is an isolated area.
> 
> Amazing work 👏🏼

Tracking this here https://github.com/orgs/apache/projects/315/views/1?pane=issue&itemId=53539080

**Pull request merged**